cms/djangoapps/contentstore/views/item.py

```diff
@@ -194,13 +201,41 @@ def xblock_handler(request, usage_key_string):
                 request.json.get('display_name'),
             )
 
-            return JsonResponse({"locator": unicode(dest_usage_key), "courseKey": unicode(dest_usage_key.course_key)})
+            return JsonResponse({'locator': unicode(dest_usage_key), 'courseKey': unicode(dest_usage_key.course_key)})
+        elif 'move_source_locator' in request.json:
```

Comment:
I would change this logic so that block moves only happen for ``PATCH`` requests. You could support patching any values for the block itself, and then if ``parent_locator`` is provided and is different then the code would also reparent the block.